test/python/circuit/test_initializer.py

```diff
@@ -446,6 +447,64 @@ def test_global_phase_1q(self):
                 actual = zero & disentangler
                 self.assertEqual(target, actual)
 
+    def test_decompose_with_int(self):
+        """Test initialize with int arg decomposes to a StatePreparation and reset"""
+        qc = QuantumCircuit(2)
+        qc.initialize(2)
+        decom_circ = qc.decompose()
+        decom_circ2 = decom_circ.decompose()
+        dag1 = circuit_to_dag(decom_circ)
+        dag2 = circuit_to_dag(decom_circ2)
+
+        self.assertEqual(len(dag1.op_nodes()), 3)
+        self.assertEqual(dag1.op_nodes()[0].name, "reset")
+        self.assertEqual(dag1.op_nodes()[1].name, "reset")
+        self.assertEqual(dag1.op_nodes()[2].name, "state_preparation")
+
+        self.assertEqual(len(dag2.op_nodes()), 3)
+        self.assertEqual(dag2.op_nodes()[0].name, "reset")
+        self.assertEqual(dag2.op_nodes()[1].name, "reset")
+        self.assertEqual(dag2.op_nodes()[2].name, "x")
```

Comment:
It wouldn't hurt to have at least one test that `StatePreparation` does essentially the same thing, but without the initial `Reset`.